packages/webapp/components/layouts/AccountLayout/Security/index.tsx

```diff
@@ -65,13 +69,27 @@ function AccountSecurityDefault({
   const { user, deleteAccount } = useContext(AuthContext);
   const [showDeleteAccount, setShowDeleteAccount] = useState(false);
   const [deletedAccount, setDeletedAccount] = useState(false);
+  const [alreadyLinkedProvider, setAlreadyLinkedProvider] = useState(false);
+  const [linkProvider, setLinkProvider] = useState(null);
   const [unlinkProvider, setUnlinkProvider] = useState(null);
   const [, setEmail] = useState<string>(null);
   const { data: userProviders } = useQuery('providers', getKratosProviders);
   const { data: settings } = useQuery('settings', () =>
     initializeKratosFlow(AuthFlow.Settings),
   );
 
+  useWindowEvents('message', async (e) => {
+    if (e.data?.flow) {
+      const flow = await getKratosSettingsFlow(AuthFlow.Settings, e.data.flow);
+      const { ui } = flow;
+      const error = ui.messages[0]?.id;
```

Comment:
We should also check the type for the `message` property as it can be sometimes just an info message for a successful event.

Reply:
Oh good point, not sure what it should do.

Reply:
Oh, I just realized you compared the ID already, and for info related ones, I think it will have a different ID. This should be good, I think.